depz/depz.py

```diff
@@ -840,6 +839,28 @@ def add_command (self, c, priority=0):
       self.commands[priority] = []
     self.commands[priority].append(c)
 
+  def queue_command (self, name, requested_on_cmd_line, f, allow_default_actions, alt_name=None):
+    if not requested_on_cmd_line and not allow_default_actions: return
+
+    funcname = getattr(f, "__name__", "")
+    if funcname.startswith("do_"):
+      funcname = funcname[3:]
+    else:
+      funcname = None
+
+    if requested_on_cmd_line and funcname: log.debug("Running function %s", funcname)
+    for i,(rname,rr) in enumerate(self.all_repos.items()):
+        if not requested_on_cmd_line:
+          if not allow_default_actions: return
+          if not App._is_default_action(name,rr) and (alt_name is None or not App._is_default_action(alt_name,rr)): return
+          log.debug("Running function %s", funcname)
```

Comment:
So the original version of this line in for_each_repo() logged when the function was actually running (before actually iterating through each repo).

Here, this will log it when it's *queued*, before anything is run.

I vote for trying to replicate the old behavior.

There are lots of ways to do this.  Like we could make a "group command" that first logged something and then executed a bunch of subcommands.  But I think maybe the easiest thing in this case is to push a simple command that just logs "Running function...", then go through the `all_repos` loop, and if that loop didn't actually push any new commands, then pop the top command (the one that emits the log message) so it doesn't log about something that doesn't actually occur.

Reply:
Good catch! Seems like the same list trick you used with `do_dump, first=[True]` should work here. I'll take a stab at that.

Reply:
Fixed! I assume you'll just squash-merge the three commits so I didn't bother myself.